ethdb/pebble/pebble.go

```diff
@@ -0,0 +1,387 @@
+	db *pebble.DB // Underlying pebble storage engine
+
+	compTimeMeter      metrics.Meter // Meter for measuring the total time spent in database compaction
+	compReadMeter      metrics.Meter // Meter for measuring the data read during compaction
+	compWriteMeter     metrics.Meter // Meter for measuring the data written during compaction
+	writeDelayNMeter   metrics.Meter // Meter for measuring the write delay number due to database compaction
+	writeDelayMeter    metrics.Meter // Meter for measuring the write delay duration due to database compaction
+	diskSizeGauge      metrics.Gauge // Gauge for tracking the size of all the levels in the database
+	diskReadMeter      metrics.Meter // Meter for measuring the effective amount of data read
+	diskWriteMeter     metrics.Meter // Meter for measuring the effective amount of data written
+	memCompGauge       metrics.Gauge // Gauge for tracking the number of memory compaction
+	level0CompGauge    metrics.Gauge // Gauge for tracking the number of table compaction in level0
+	nonlevel0CompGauge metrics.Gauge // Gauge for tracking the number of table compaction in non0 level
+	seekCompGauge      metrics.Gauge // Gauge for tracking the number of table compaction caused by read opt
+
+	quitLock sync.Mutex      // Mutex protecting the quit channel access
+	quitChan chan chan error // Quit channel to stop the metrics collection before closing the database
+
+	log log.Logger // Contextual logger tracking the database path
+}
+
+// New returns a wrapped LevelDB object. The namespace is the prefix that the
+// metrics reporting should use for surfacing internal stats.
+func New(file string, cache int, handles int, namespace string) (*Database, error) {
+	// Ensure we have some minimal caching and file guarantees
+	if cache < minCache {
+		cache = minCache
+	}
+	if handles < minHandles {
+		handles = minHandles
+	}
+	logger := log.New("database", file)
+	logger.Info("Allocated cache and file handles", "cache", common.StorageSize(cache*1024*1024), "handles", handles)
+
+	// Open the db and recover any potential corruptions
+	db, err := pebble.Open(file, &pebble.Options{
+		// SStable and WALs' sync rate. The default rate is 512KB.
+		// It's worthwhile to tune the performance by this setting.
+		BytesPerSync: 0, // A larger number maybe is better?
+		Cache:        pebble.NewCache(int64(cache * 1024 * 1024 / 2)),
+		MaxOpenFiles: handles,
+		// The size of memory table(as well as the write buffer).
+		// Note, there may have more than two memory tables in the system.
+		// MemTableStopWritesThreshold can be configured to avoid the memory abuse.
+		MemTableSize: cache * 1024 * 1024 / 4,
+		// The default compaction concurrency(1 thread),
+		// Here use all available CPUs for faster compaction.
+		MaxConcurrentCompactions: runtime.NumCPU(),
+		// The default sstable format(rocksdb compatible).
+		// Pick one for selecting leveldb compatible format.
+		TableFormat: pebble.TableFormatRocksDBv2,
```

Comment:
We can drop this, it's not configurable https://github.com/cockroachdb/pebble/issues/915